### PR TITLE
Feature/create db

### PR DESCRIPTION
**Pull request by idob2**

### CODE REVIEW

File: src/controllers/v1/chef_controller.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  try {
+    const chef = await findAllChefDishes(chefId);
+    if (!chef) {
+      res.status(404).json({ error: "Unable to fetch chef" });
+    }
+    const allDishes: any[] = [];
+    chef?.restaurants.forEach((restaurant: any) => {
+      allDishes.push(...restaurant.dishes);
+    });
+    res.json(allDishes);
```
Comment (GuyLeiferMoveo) [inline src/controllers/v1/chef_controller.ts]: do it with map array prototype inline instead forEach prototype with two lines

File: src/controllers/v1/chef_controller.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ error: "Unable to fetch the chef." });
+  }
+};
+
+
+const postChef = async (req: Request, res: Response) => {
+  const { name, image, description, restaurants } = req.body;
+  console.log(req.body);
```
Comment (GuyLeiferMoveo) [inline src/controllers/v1/chef_controller.ts]: remove unnecessary logs

File: src/handlers/chef_handler.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  return chef;
+};
+const addChef = async (
+  name: string,
+  image: string,
+  description: string,
+  restaurants: string
+) => {
+  const newChef = new Chef({
+    _id: 12,
```
Comment (GuyLeiferMoveo) [inline src/handlers/chef_handler.ts]: why the id is hardcoded?

File: src/controllers/v1/chef_controller.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+//     console.log(value);
+//     const chefs = await findChefByQuery(value);
+//     res.json(chefs);
+//   } catch (error) {
+//     console.log(error);
+//     res.status(500).json({ error: "Unable to fetch chefs." });
+//   }
+// }
+const getChefByID = async (req: Request, res: Response) => {
+  const chefId = req.params.id;
```
Comment (GuyLeiferMoveo) [inline src/controllers/v1/chef_controller.ts:22]: could it be undefined?

File: src/controllers/v1/chef_controller.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  try {
+    const chef = await findChefById(chefId);
```
Comment (GuyLeiferMoveo) [inline src/controllers/v1/chef_controller.ts:24]: and if chefId could be undefined, what will happen?

File: src/controllers/v1/chef_controller.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    if (!chef) {
+      return res.status(404).json({ error: "Chef not found." });
+    }
+    res.json(chef);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({ error: "Unable to fetch the chef." });
```
Comment (GuyLeiferMoveo) [inline src/controllers/v1/chef_controller.ts]: throw the error.message

File: src/controllers/v1/chef_controller.ts (hunk 4 above)
Comment (idob2) [reply src/controllers/v1/chef_controller.ts:35]: mistake

File: src/handlers/chef_handler.ts (hunk 3 above)
Comment (idob2) [reply src/handlers/chef_handler.ts]: mistake

File: src/controllers/v1/chef_controller.ts (hunk 4 above)
Comment (idob2) [reply src/controllers/v1/chef_controller.ts:22]: yes

File: src/controllers/v1/chef_controller.ts (hunk 5 above)
Comment (idob2) [reply src/controllers/v1/chef_controller.ts:24]: it will throw an error, I this that what we should expect

**Pull request merged**